### PR TITLE
fix(notification): fit long words

### PR DESCRIPTION
**Pull request by JessieTeng89**
Fixes https://github.com/argoproj/argo-ui/issues/512
It is a layout issue when there is a long word in the error message toast popup, so added a style in that component to fix the issue.

Screenshots:
before fix:
image

after fix:
image

### CODE REVIEW

File: src/components/notifications/notifications.tsx
Comment (agilgur5) [inline src/components/notifications/notifications.tsx:61]:
```suggestion
                    overflowWrap: "break-word"
```

[`break-word`](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap) might be more legible. but test it out and make sure

Comment (agilgur5) [reply src/components/notifications/notifications.tsx:61]: [Looks like](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap#comparing_overflow-wrap_word-break_and_hyphens) the main difference is that it will add a newline before the long word. I'm honestly not sure if that's more legible or not 😅

Comment (JessieTeng89) [reply src/components/notifications/notifications.tsx:61]: > [Looks like](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap#comparing_overflow-wrap_word-break_and_hyphens) the main difference is that it will add a newline before the long word. I'm honestly not sure if that's more legible or not 😅
> 
> I think it would avoid the "denie-d" in your screenshot and would make it on one line though? That would be better for shorter words that _could_ fit on one line then

Thanks for your suggestion, you are correct, it is much better to use `break-word` instead of `break-all` so that it only breaks the long words.
But I found it is only working with `word-break: break-word;` and it is not working with `overflow-wrap: break-word`, I found someone is facing the same issue with this component, I can change my PR to `word-break: break-word;` if it is good to you.
https://github.com/fkhadra/react-toastify/issues/112

After using `word-break: break-word;`:
image

Comment (agilgur5) [reply src/components/notifications/notifications.tsx:61]: `word-break: break-word` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break#browser_compatibility). It appears to still be supported, but could be removed from browsers, so is not necessarily stable. That's why I suggested `overflow-wrap: break-word`, which is supported.

Strange that it didn't work for you, I wonder if there's some other CSS needed on the `display` or `overflow`? 🤔 
Or if it was overridden by something in the existing CSS?

Another option is `overflow-x: scroll`, which is not the nicest, but works relatively well with touch pads at least.

Comment (JessieTeng89) [reply src/components/notifications/notifications.tsx]: > `word-break: break-word` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break#browser_compatibility). It appears to still be supported, but could be removed from browsers, so is not necessarily stable. That's why I suggested `overflow-wrap: break-word`, which is supported.
> 
> Strange that it didn't work for you, I wonder if there's some other CSS needed on the `display` or `overflow`? 🤔 Or if it was overridden by something in the existing CSS?
> 
> Some alternatives include:
> 
> * `overflow-x: scroll`, which is not the nicest, but works relatively well with touch pads at least.
> * [`min-content`](https://developer.mozilla.org/en-US/docs/Web/CSS/min-content) sizing or similar, which would stretch the notification, potentially too much

Hi @agilgur5 
I did some changes to use recommended css override way to fix this issue, could you please help to review?

**Comment by JessieTeng89** [comment]
@crenshaw-dev Kindly help to review this PR

**Comment by JessieTeng89** [comment]
Hi @crenshaw-dev 
Can I know how to get this PR approved or if any other actions needed before merging it?
Thanks

**Comment by JessieTeng89** [comment]
Hi @agilgur5
A gentle follow up with this PR, could you please help to review it?

**Comment by agilgur5** [comment]
Yes, I had been planning to; I'm not sure why you asked several other uninvolved people [for review](https://github.com/argoproj/argo-ui/pull/546#issuecomment-2068031295) a day after you [made changes](https://github.com/argoproj/argo-ui/pull/546#discussion_r1573288510), which was a full 2 weeks after [changes were requested](https://github.com/argoproj/argo-ui/pull/546#discussion_r1553002577).

Please be patient, especially if you yourself do not respond immediately -- it would not be surprising if a maintainer took as long as you did to respond if not longer. And please interact with maintainers who already reviewed your PR, instead of asking others, which is counter-productive.

File: src/components/notifications/notifications.scss
Comment (agilgur5) [inline src/components/notifications/notifications.scss:10]: I would not recommend overriding the component's styles. There are simpler ways of achieving this. This is also a global style, so it affects all usage of `Toastify`, not just the ones in the `Notifications` component. It should be scoped to the component if you were to do something like this.

File: src/components/notifications/notifications.tsx
Comment (agilgur5) [reply src/components/notifications/notifications.tsx]: I'd prefer to use `word-break: break-word;` than overriding the component's styles, it is less complicated.

But a simpler way to achieve this is to just add `max-width: 240px` in addition to the `overflow-wrap: 'break-word';`. You can put them on the `div` itself, instead of passing to the `toastMethod`.

I did confirm this works:
Screenshot 2024-06-10 at 11 06 08 PM

Since the toast's width is never changed, we can reliably set a fixed value for `max-width`. (although increasing the width of the toast might be a good idea for this problem)

**Review by agilgur5 — CHANGES_REQUESTED** [review]
There are simpler ways of achieving this without overriding the component's styles, which can cause more problems and is not very reliable

File: src/components/notifications/notifications.tsx
Comment (agilgur5) [inline src/components/notifications/notifications.tsx:24]: I can't push to your branch and GH can't comment on unchanged lines, so here's what I was suggesting.

Remove this CSS file and import, instead, on line 50/52 below of the `div`, add:
```tsx
               <div onClick={(e) => {
                 // ... elided ... don't change anything with the `onClick`
                }} style={{
                    // fit long words
                    overflowWrap: 'break-word',
                    maxWidth: '240px'
                }}>
```

Comment (JessieTeng89) [reply src/components/notifications/notifications.tsx:24]: I totally agree, Thanks a lot, I will make the changes and let you know

File: src/components/notifications/notifications.tsx
Comment (agilgur5) [inline src/components/notifications/notifications.tsx:51]:
```suggestion
                    // fit long words by wrapping them instead of overflowing past the width
                    overflowWrap: 'break-word',
```

add the comment I mentioned [before](https://github.com/argoproj/argo-ui/pull/546#discussion_r1634122451)

Comment (agilgur5) [reply src/components/notifications/notifications.tsx:51]: I unfortunately can't commit that myself via suggestion as a maintainer normally can, I'm guessing because of Fidelity's permissions

Comment (JessieTeng89) [reply src/components/notifications/notifications.tsx:51]: Thanks, I will add the comment

Comment (JessieTeng89) [reply src/components/notifications/notifications.tsx:52]: Hi @agilgur5 
Just added explanatory comment as suggested

**Review by agilgur5 — APPROVED** [review]
LGTM. Thanks for noting and fixing this issue

**Comment by gillesbouvier-qz** [comment]
Did this change add the following CSS?
```CSS
.Toastify__toast-body > div:last-child {
  word-break: break-word;
  ...
}
```

If so, it is not working as expected as it is breaking short words too. Did you mean to use this instead?
```CSS
    overflow-wrap: break-word;
```

**Comment by agilgur5** [comment]
No, you can see the diff here, which is all of 4 LoC of an in-line style that uses `overflow-wrap: break-word`

**Pull request merged**
